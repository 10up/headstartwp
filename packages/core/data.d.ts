export * from './dist/mjs/data';
