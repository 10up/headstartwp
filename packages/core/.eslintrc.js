module.exports = {
	settings: {
		'import/core-modules': ['@10up/headless-core', '@10up/headless-core/utils'],
	},
	rules: {
		'jsdoc/require-returns-type': 0,
	},
};
