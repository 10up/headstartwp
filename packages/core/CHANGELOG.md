# @10up/headless-core

## 0.3.0

### Minor Changes

- 9b992b9: Throw an error when trying to access data from useFetch hooks when the data is not yet avaliable.

## 0.3.0-next.0

### Minor Changes

- 9b992b9: Throw an error when trying to access data from useFetch hooks when the data is not yet avaliable.

## 0.2.1

### Patch Changes

- e9c7ef0: remove emotion dep from core package

## 0.2.1-next.0

### Patch Changes

- e9c7ef0: remove emotion dep from core package

## 0.2.0

### Minor Changes

- a68a9b7: Release stable version
