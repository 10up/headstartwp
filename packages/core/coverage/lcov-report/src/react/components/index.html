
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/react/components</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> src/react/components</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">76.4% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>68/89</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">46.51% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>20/43</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">58.82% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>10/17</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">75.29% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>64/85</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="BlocksRenderer.tsx"><a href="BlocksRenderer.tsx.html">BlocksRenderer.tsx</a></td>
	<td data-value="87.23" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 87%"></div><div class="cover-empty" style="width: 13%"></div></div>
	</td>
	<td data-value="87.23" class="pct high">87.23%</td>
	<td data-value="47" class="abs high">41/47</td>
	<td data-value="73.07" class="pct medium">73.07%</td>
	<td data-value="26" class="abs medium">19/26</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="87.23" class="pct high">87.23%</td>
	<td data-value="47" class="abs high">41/47</td>
	</tr>

<tr>
	<td class="file high" data-value="HtmlDecoder.tsx"><a href="HtmlDecoder.tsx.html">HtmlDecoder.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file medium" data-value="Link.tsx"><a href="Link.tsx.html">Link.tsx</a></td>
	<td data-value="66.66" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 66%"></div><div class="cover-empty" style="width: 34%"></div></div>
	</td>
	<td data-value="66.66" class="pct medium">66.66%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	</tr>

<tr>
	<td class="file low" data-value="Menu.tsx"><a href="Menu.tsx.html">Menu.tsx</a></td>
	<td data-value="44" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 44%"></div><div class="cover-empty" style="width: 56%"></div></div>
	</td>
	<td data-value="44" class="pct low">44%</td>
	<td data-value="25" class="abs low">11/25</td>
	<td data-value="5.88" class="pct low">5.88%</td>
	<td data-value="17" class="abs low">1/17</td>
	<td data-value="14.28" class="pct low">14.28%</td>
	<td data-value="7" class="abs low">1/7</td>
	<td data-value="41.66" class="pct low">41.66%</td>
	<td data-value="24" class="abs low">10/24</td>
	</tr>

<tr>
	<td class="file high" data-value="SafeHtml.tsx"><a href="SafeHtml.tsx.html">SafeHtml.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file high" data-value="index.ts"><a href="index.ts.html">index.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2023-08-16T12:58:09.325Z
            </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    