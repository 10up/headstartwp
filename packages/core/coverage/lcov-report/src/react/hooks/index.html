
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/react/hooks</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> src/react/hooks</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">92.68% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>152/164</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">77.55% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>76/98</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>29/29</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">92.68% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>152/164</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="useFetch.ts"><a href="useFetch.ts.html">useFetch.ts</a></td>
	<td data-value="93.33" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 93%"></div><div class="cover-empty" style="width: 7%"></div></div>
	</td>
	<td data-value="93.33" class="pct high">93.33%</td>
	<td data-value="30" class="abs high">28/30</td>
	<td data-value="72.72" class="pct medium">72.72%</td>
	<td data-value="11" class="abs medium">8/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="93.33" class="pct high">93.33%</td>
	<td data-value="30" class="abs high">28/30</td>
	</tr>

<tr>
	<td class="file high" data-value="useFetchAppSettings.ts"><a href="useFetchAppSettings.ts.html">useFetchAppSettings.ts</a></td>
	<td data-value="85.71" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 85%"></div><div class="cover-empty" style="width: 15%"></div></div>
	</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="14" class="abs high">12/14</td>
	<td data-value="77.77" class="pct medium">77.77%</td>
	<td data-value="9" class="abs medium">7/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="14" class="abs high">12/14</td>
	</tr>

<tr>
	<td class="file high" data-value="useFetchAuthorArchive.ts"><a href="useFetchAuthorArchive.ts.html">useFetchAuthorArchive.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="71.42" class="pct medium">71.42%</td>
	<td data-value="7" class="abs medium">5/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

<tr>
	<td class="file high" data-value="useFetchPost.ts"><a href="useFetchPost.ts.html">useFetchPost.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="14" class="abs high">14/14</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="12" class="abs high">10/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="14" class="abs high">14/14</td>
	</tr>

<tr>
	<td class="file high" data-value="useFetchPostOrPosts.ts"><a href="useFetchPostOrPosts.ts.html">useFetchPostOrPosts.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="19" class="abs high">19/19</td>
	<td data-value="76.47" class="pct medium">76.47%</td>
	<td data-value="17" class="abs medium">13/17</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="19" class="abs high">19/19</td>
	</tr>

<tr>
	<td class="file high" data-value="useFetchPosts.ts"><a href="useFetchPosts.ts.html">useFetchPosts.ts</a></td>
	<td data-value="87.5" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 87%"></div><div class="cover-empty" style="width: 13%"></div></div>
	</td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="40" class="abs high">35/40</td>
	<td data-value="80.95" class="pct high">80.95%</td>
	<td data-value="21" class="abs high">17/21</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="40" class="abs high">35/40</td>
	</tr>

<tr>
	<td class="file high" data-value="useFetchSearch.ts"><a href="useFetchSearch.ts.html">useFetchSearch.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="18" class="abs high">18/18</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="10" class="abs high">8/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="18" class="abs high">18/18</td>
	</tr>

<tr>
	<td class="file high" data-value="useFetchTerms.ts"><a href="useFetchTerms.ts.html">useFetchTerms.ts</a></td>
	<td data-value="85.71" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 85%"></div><div class="cover-empty" style="width: 15%"></div></div>
	</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="14" class="abs high">12/14</td>
	<td data-value="88.88" class="pct high">88.88%</td>
	<td data-value="9" class="abs high">8/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="14" class="abs high">12/14</td>
	</tr>

<tr>
	<td class="file high" data-value="util.ts"><a href="util.ts.html">util.ts</a></td>
	<td data-value="85.71" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 85%"></div><div class="cover-empty" style="width: 15%"></div></div>
	</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="7" class="abs high">6/7</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="7" class="abs high">6/7</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2023-08-16T12:58:09.325Z
            </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    