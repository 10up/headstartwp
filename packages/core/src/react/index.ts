export * from './components';
export * from './provider';
export * from './hooks';
export * from './blocks';
export * from './utils';
export * from '../dom';
