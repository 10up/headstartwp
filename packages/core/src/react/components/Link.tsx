import { PropsWithChildren } from 'react';

interface RawLinkProps {
	href: string;
	children: React.ReactNode;
}

export const RawLink = ({ children, href, ...props }: PropsWithChildren<RawLinkProps>) => {
	return (
		<a href={href} {...props}>
			{children}
		</a>
	);
};
