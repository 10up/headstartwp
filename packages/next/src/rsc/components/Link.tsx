'use client';

import { removeSourceUrl } from '@headstartwp/core';
import { useSettings } from '@headstartwp/core/react';
import { FC } from 'react';
import NextLink from 'next/link.js';

type LinkProps = React.ComponentProps<typeof NextLink>;

export const Link: FC<LinkProps> = ({ children, href, ...props }) => {
	const settings = useSettings();

	if (typeof settings === 'undefined') {
		throw new Error('The Link component must be used within the SettingsProvider');
	}

	const link =
		typeof href === 'string'
			? removeSourceUrl({
					link: href,
					backendUrl: settings.sourceUrl || '',
					publicUrl: settings.hostUrl ?? '/',
				})
			: href;

	return (
		<NextLink href={link} {...props}>
			{children}
		</NextLink>
	);
};
