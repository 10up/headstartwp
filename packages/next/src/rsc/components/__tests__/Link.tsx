import { render } from '@testing-library/react';
import React from 'react';
import { SettingsProvider } from '@headstartwp/core/react';
import { Link } from '../Link';

describe('Link', () => {
	it('shound replace internal links with next/link', () => {
		const { container } = render(
			<SettingsProvider
				settings={{
					sourceUrl: 'http://wpadmin.com',
					hostUrl: 'http://domain.com',
				}}
			>
				<Link href="http://wpadmin.com/post-name" />
			</SettingsProvider>,
		);

		expect(container).toMatchSnapshot();
	});
});
