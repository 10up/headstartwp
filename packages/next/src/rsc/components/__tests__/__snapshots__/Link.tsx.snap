// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link shound replace internal links with next/link 1`] = `
<div>
  <a
    href="/post-name"
  />
</div>
`;
